crates/matrix-sdk-base/src/notification_settings.rs

```diff
@@ -19,10 +19,15 @@ use serde::{Deserialize, Serialize};
 /// Enum representing the push notification modes for a room.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
 pub enum RoomNotificationMode {
-    /// Receive notifications for all messages.
+    /// Receive remote and in-app notifications for all messages.
     AllMessages,
-    /// Receive notifications for mentions and keywords only.
-    MentionsAndKeywordsOnly,
+    /// Receive remote and in-app notifications for mentions and keywords only.
+    MentionsAndKeywordsOnly {
+        /// If true, receive in-app-only notifications (no pushes) for other
+        /// room messages. Otherwise, mute all other room messages.
+        #[cfg(feature = "unstable-msc3768")]
+        notify_in_app: bool,
+    },
```

Comment:
Sorry to go back and forth with this, but I've realized that since this `enum` is `Deserialize`, this would need a data migration, at the risk of making `RoomInfo` deserialization fail (thus loading of initial rooms fail at start).

But thanks to the nicely split commits (thanks!), I think you can revert deef9610d75d0d1b88423a800a1cabb1f74f29e2 and we should be back to the previous place. Then we'd rename `MentionsAndKeywordsOnlyTheRestInApp` to something like `PushMentionsAndKeywordsOnly`, and we'll be good, for the time being, in terms of client-side implementation.

Reply:
No worries. Was an easy revert. Fixed with b23bc278a701c752e9a71ea3dbee7885f9705cc0 and e6bd99f296e7ad03b0b1b307223c2c2bed0f917f.